### PR TITLE
fix: Fixed missing parentheses after `super` calls.

### LINKED ISSUE #26360: Missing Parentheses after `super` calls at few places.
In the following places `super` is used without `parentheses`, which can lead to unexpected behavior in Python 3.
https://github.com/unifyai/ivy/blob/86afb43b97dab3ca2cd10f24bbe631a6cbfcdc6f/ivy/data_classes/container/base.py#L4008
https://github.com/unifyai/ivy/blob/86afb43b97dab3ca2cd10f24bbe631a6cbfcdc6f/ivy/data_classes/container/base.py#L4035
https://github.com/unifyai/ivy/blob/86afb43b97dab3ca2cd10f24bbe631a6cbfcdc6f/ivy_tests/test_ivy/helpers/globals.py#L66

`Parentheses` should be added after super to ensure correct behavior. I found these using [pylint](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/super-without-brackets.html).

### PR DESCRIPTION
**Pull request by Sai-Suraj-27**
# PR Description
In the following places `super` is used without `parentheses`, which can lead to unexpected behavior in Python 3.
https://github.com/unifyai/ivy/blob/86afb43b97dab3ca2cd10f24bbe631a6cbfcdc6f/ivy/data_classes/container/base.py#L4008
https://github.com/unifyai/ivy/blob/86afb43b97dab3ca2cd10f24bbe631a6cbfcdc6f/ivy/data_classes/container/base.py#L4035
https://github.com/unifyai/ivy/blob/86afb43b97dab3ca2cd10f24bbe631a6cbfcdc6f/ivy_tests/test_ivy/helpers/globals.py#L66

`Parentheses` should be added after super to ensure correct behavior. I found these using [pylint](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/super-without-brackets.html) and fixed them.

## Related Issue
Close #26360

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [ ] Did you follow the steps we provided?

### Socials
https://twitter.com/Sai_Suraj_27

### CODE REVIEW

File: ivy/data_classes/container/base.py [hunk 1]
```diff
@@ -4005,7 +4005,7 @@ def _add_newline(str_in):
         return new_dict
 
     def __dir__(self):
-        return list(super.__dir__(self)) + list(self.keys())
+        return list(super().__dir__(self)) + list(self.keys())
```
Comment (sherry30) [inline ivy/data_classes/container/base.py:4008]: You dont pass in self when using `super()`

**Review by sherry30 — CHANGES_REQUESTED** [review]
one requested change

File: ivy/data_classes/container/base.py (hunk 1 above)
Comment (Sai-Suraj-27) [reply ivy/data_classes/container/base.py]: @sherry30 sir, I made the requested changes. Can you please check now, Thank you.

**Review by CatB1t — APPROVED** [review]
lgtm!

**Comment by Sai-Suraj-27** [comment]
@sherry30 sir, can you look into this PR? its been a week since I made the requested changes. Please look into this when you are free, Thank you.

**Pull request merged**